projects/opendr_ws/src/perception/scripts/object_detection_3d_voxel.py

```diff
@@ -80,39 +79,67 @@ def callback(self, data):
 
         # Convert detected boxes to ROS type and publish
         ros_boxes = self.bridge.to_ros_boxes_3d(detection_boxes, classes=["Car", "Van", "Truck", "Pedestrian", "Cyclist"])
-        if self.detection_publisher is not None:
-            self.detection_publisher.publish(ros_boxes)
-            rospy.loginfo("Published detection boxes")
+        self.detection_publisher.publish(ros_boxes)
+        rospy.loginfo("Published detection boxes")
+
+    def listen(self):
+        """
+        Start the node and begin processing input data.
+        """
+        rospy.init_node('voxel_detection_3d', anonymous=True)
+        rospy.Subscriber(self.input_point_cloud_topic, ROS_PointCloud, self.callback, queue_size=1, buff_size=10000000)
 
-if __name__ == "__main__":
-    # Automatically run on GPU/CPU
-    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        rospy.loginfo("Object Detection 3D Voxel Node started.")
+        rospy.spin()
 
-    # initialize ROS node
-    rospy.init_node("opendr_voxel_detection_3d", anonymous=True)
-    rospy.loginfo("Voxel Detection 3D node started")
 
-    model_name = rospy.get_param("~model_name", "tanet_car_xyres_16")
-    model_config_path = rospy.get_param(
-        "~model_config_path", os.path.join(
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-n", "--model_name", help="Name of the trained model",
+                        type=str, default="tanet_car_xyres_16")
```

Comment:
Added choices argument, it seems redundant but it shows up when calling help and avoids users providing wrong values.

```suggestion
    parser.add_argument("-n", "--model_name", help="Name of the trained model",
                        type=str, default="tanet_car_xyres_16", choices=["tanet_car_xyres_16"])
```